### DISCUSSION TITLE
No context swaps for llama-server

### DISCUSSION THREAD

**Discussion by Tureti** [ggml-org/llama.cpp#8310] [Q&A] [answered]
Hey, I currently use llama-server with Open-WebUI. When the context is long, I have to wait a long time before it starts writing an answer every time I ask something. This is not the case with llama-cli in conversation mode as follow up questions don't make the model reprocess the whole prompt.
Is there an option to enable the same thing for llama-server? Or is it not possible?

Edit: https://github.com/ggerganov/llama.cpp/pull/4347 seems to be the fix but can I specify cache_prompt=True when starting the server or do I need to send it with every request?

**Comment by Tureti** [comment]
For now I just changed the default value for cache_prompt to true and rebuild it since I can't find which parameter to pass to llama-server for it.

**Reply by ggerganov** [reply, marked as answer]
You need to send the `cache_prompt` with every request
